src/thunderbots/software/ai/hl/stp/tactic/tactic.h

```diff
@@ -17,24 +17,85 @@ class Tactic
 {
    public:
     /**
-     * Returns the next Intent that should be run by the Robot assigned to this Tactic
-     * in order to work towards achieving the Tactic's objective
-     * @param world The current state of the world
-     * @param robot The Robot assigned to this Tactic
-     * @return The Intent that the Robot should run in order to work towards the Tactic's
-     * objective
+     * Creates a new Tactic with the given robot
+     *
+     * @param robot The robot that should perform this Tactic
      */
-    virtual std::unique_ptr<Intent> getNextIntent(const World& world,
-                                                  const Robot& robot) = 0;
+    explicit Tactic(const Robot& robot);
 
     /**
-     * Returns the Robot that the Tactic would prefer to use
-     * @param available_robots The list of available Robots that the Tactic can choose
-     * from
-     * @return The Robot that the Tactic prefers to use
+     * Returns true if the Tactic is done and false otherwise
+     *
+     * @return true if the Tactic is done and false otherwise
      */
-    virtual Robot selectRobot(const World& world,
-                              const std::vector<Robot>& available_robots) = 0;
+    bool done() const;
+
+    /**
+     * Evaluates the given Robot and returns a score indicating how optimal it would be
+     * for that Robot to perform this Tactic. A lower score indicates a more optimal
+     * robot, and returned score values must be >= 0.
```

Comment:
Can we make this a value in `[0,1]` so we have a concrete range?

Reply:
Done

Reply:
nope

Reply:
yup